x-pack/plugins/lists/common/schemas/request/update_exception_list_item_schema.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable @typescript-eslint/camelcase */
+
+import * as t from 'io-ts';
+
+import {
+  ItemId,
+  Tags,
+  _Tags,
+  _tags,
+  comment,
+  description,
+  exceptionListItemType,
+  id,
+  meta,
+  name,
+  tags,
+} from '../common/schemas';
+import { Identity, RequiredKeepUndefined } from '../../types';
+import { DefaultEntryArray, DefaultUuid } from '../types';
+import { EntriesArray } from '../types/entries';
+
+export const updateExceptionListItemSchema = t.intersection([
+  t.exact(
+    t.type({
+      description,
+      name,
+      type: exceptionListItemType,
+    })
+  ),
+  t.exact(
+    t.partial({
+      _tags, // defaults to empty array if not set during decode
+      comment, // defaults to undefined if not set during decode
+      entries: DefaultEntryArray, // defaults to empty array if not set during decode
+      id, // defaults to undefined if not set during decode
+      item_id: DefaultUuid, // defaults to GUID (uuid v4) if not set during decode
```

Comment:
I'm still wrapping my head around our usage of `io-ts` so forgive me if I'm reading this incorrectly... but this looks like we're giving the user the ability to update the `item_id` for an already-existing list item... and if they provide the key, but its value is undefined, then we'll create a new uuid? I'd be inclined to think this should just be a string, or be left out entirely (i.e. don't let the user change it once it's been set).

Reply:
I think having a `DefaultUuid` here is misleading and a bug for updates. I will change it to default to being `undefined`. That would be consistent with the rules API and other places.

If a user wants to update an exception list, exception list item, etc... they can either send in the regular `id` which is saved object auto-generated and not user changeable/controllable by the user. However, if the user wants to update based on their controlled extern type of id (list_id/item_id/etc..) they are allowed to update an exception list or exception list item based on that particular extern type of id.

We do not allow the user to update/change their list_id/item_id which I agree wouldn't make any sense. I will remove this `DefaultUuid` here before checking in.

Reply:
Okay great, I was just making sure I understood this. That sounds like the right thing to do. Thanks!